core/trino-main/src/main/java/io/trino/execution/executor/MultilevelSplitQueue.java

```diff
@@ -304,6 +313,19 @@ public static int computeLevel(long threadUsageNanos)
         return LEVEL_THRESHOLD_SECONDS.length - 1;
     }
 
+    public DistributionStat getWaitingSplitsSizeDistribution()
+    {
+        return waitingSplitsSizeDistribution;
+    }
+
+    private void recordWaitingSplitQueueSize()
+    {
+        long now = ticker.read();
+//        this.waitingSplitsSize.add(waitingSplits.size()size, now - this.lastTimeSplitSizeWasSampled); // Waits for airlift release
+        this.waitingSplitsSizeDistribution.add(this.size());
```

Comment:
does it makes sense to add just `waitingSplitsSizeDistribution` in this PR and have another one for `waitingSplitsSize`?

Btw what class is `waitingSplitsSize`?

Reply:
`waitingSplitsSize` is same type as `waitingSplitsSizeDistribution`. I've changed the name of this field and it was not refactored becasue it is commented for now.